app/models/dataset_record/darwin_core/occurrence.rb

```diff
@@ -211,6 +211,17 @@ def import(dwc_data_attributes = {})
 
         # TODO: If all attributes are equal assume it is the same event and share it with other specimens? (eventID is an alternate method to detect duplicates)
         if collecting_event
+          # if tags have been specified to be added, update the collecting event
+          if attributes[:collecting_event][:tags_attributes]
+            # get list of preexisting tags, exclude them from update
+            current_tags = collecting_event.tags.pluck(:keyword_id).to_set
```

Comment:
I missed part of the conversation in today's meeting, but is preserving current values intentional or this is related to the validation errors you mentioned?